Sources/TuistKit/Commands/TuistCommand.swift

```diff
@@ -126,31 +126,50 @@ public struct TuistCommand: AsyncParsableCommand {
         do {
             try await executeCommand()
             outputCompletion(logFilePath: logFilePath, shouldOutputLogFilePath: logFilePathDisplayStrategy == .always)
-        } catch let error as FatalError {
-            self.outputCompletion(logFilePath: logFilePath, shouldOutputLogFilePath: true, beforeLogsLine: {
-                errorHandler.fatal(error: error)
-            })
-            _exit(exitCode(for: error).rawValue)
-        } catch let error as ClientError where error.underlyingError is ServerClientAuthenticationError {
-            ServiceContext.current?.ui?
-                // swiftlint:disable:next force_cast
-                .error(.alert("\((error.underlyingError as! ServerClientAuthenticationError).description)"))
-            outputCompletion(logFilePath: logFilePath, shouldOutputLogFilePath: true)
-            _exit(exitCode(for: error).rawValue)
         } catch {
-            if let parsedError {
-                handleParseError(parsedError)
-            }
+            onError(parsingError ?? error, isParsingError: parsingError != nil, logFilePath: logFilePath)
+        }
+    }
 
-            // Exit cleanly
-            if exitCode(for: error).rawValue == 0 {
-                exit(withError: error)
-            } else {
-                errorHandler.fatal(error: UnhandledError(error: error))
-                outputCompletion(logFilePath: logFilePath, shouldOutputLogFilePath: true)
-                _exit(exitCode(for: error).rawValue)
+    private static func onError(_ error: Error, isParsingError: Bool, logFilePath: AbsolutePath) {
+        var errorAlertMessage: TerminalText?
+        var errorAlertNextSteps: [TerminalText] = [
+            "If the error is actionable, address it",
+            "If the error is not actionable, let's discuss it in the \(.link(title: "Troubleshooting & how to", href: "https://community.tuist.dev/c/troubleshooting-how-to/6"))",
+            "If you are very certain it's a bug, \(.link(title: "file an issue", href: "https://github.com/tuist/tuist"))",
+        ]
+        let exitCode = exitCode(for: error).rawValue
+
+        if let clientError = error as? ClientError, clientError.underlyingError is ServerClientAuthenticationError {
```

Comment:
```suggestion
        if let clientError = error as? ClientError, let underlyingServerClientError = clientError.underlyingError as? ServerClientAuthenticationError {
```

That way we can skip some force unwrapping